src/modules/AutoInstallRules.rb

```diff
@@ -874,94 +874,78 @@ def GetRules
       end
     end
 
+    #TODO: Move the responsibility of merging profiles to a specific class
+    # removing also the duplication of code between this module and the
+    # AutoinstClass one.
+
+    MERGE_CMD = "/usr/bin/xsltproc".freeze
+    MERGE_DEFAULTS = "--novalid --maxdepth 10000 --param replace \"'false'\"".freeze
+    MERGE_XSLT_PATH = "/usr/share/autoinstall/xslt/merge.xslt".freeze
+
+    # Merges the given profiles
+    #
+    # @param base_profile [String] the base profile file path
+    # @param with [String] the profile to be merged file path
+    # @param to [String] the resulting control file path
+    # @return [Hash] stdout and stderr output
+    def merge_profiles(base_profile, with, to)
+      dontmerge_str = ""
+      AutoinstConfig.dontmerge.each_with_index do |dm, i|
+        dontmerge_str << " --param dontmerge#{i+1} \"'#{dm}'\""
+      end
+      merge_command =
+        "#{MERGE_CMD} #{MERGE_DEFAULTS}" \
+        "#{dontmerge_str} --param with \"'#{with}'\" " \
+        "--output \"#{to}\" " \
+        "#{MERGE_XSLT_PATH} #{base_profile}"
+
+      out = SCR.Execute(path(".target.bash_output"), merge_command, {})
+      log.info("Merge command: #{merge_command}")
+      log.info("Merge stdout: #{out["stdout"]}, stderr: #{out["stderr"]}")
+      out
+    end
 
     # Merge Rule results
     # @param [String] result_profile the resulting control file path
     # @return [Boolean] true on success
     def Merge(result_profile)
-      tmpdir = AutoinstConfig.tmpDir
+      base_profile = File.join(AutoinstConfig.tmpdir, "base_profile.xml")
+      merge_profile = File.join(AutoinstConfig.tmpdir, "result.xml")
       ok = true
       skip = false
       error = false
-
-      base_profile = Ops.add(tmpdir, "/base_profile.xml")
-
-      Builtins.foreach(@tomerge) do |file|
-        Builtins.y2milestone("Working on file: %1", file)
-        current_profile = Ops.add(
-          Ops.add(AutoinstConfig.local_rules_location, "/"),
-          file
-        )
+      @tomerge.each do |file|
+        log.info("Working on file: %1", file)
+        current_profile = File.join(AutoinstConfig.local_rules_location, file)
         if !skip
-          if !XML_cleanup(current_profile, Ops.add(tmpdir, "/base_profile.xml"))
-            Builtins.y2error("Error reading XML file")
+          if !XML_cleanup(current_profile, base_profile)
+            log.error("Error reading XML file")
             message = _(
               "The XML parser reported an error while parsing the autoyast profile. The error message is:\n"
             )
-            message = Ops.add(message, XML.XMLError)
-            Popup.Error(message)
+            Popup.Error(message + XML.XMLError)
```

Comment:
You may want to use the same popup API here (see https://github.com/yast/yast-autoinstallation/pull/450/files).

Reply:
Thanks for pointing to it